src/features/machine/selectors.js

```diff
@@ -1,47 +1,98 @@
 import { createSelector } from 'reselect'
 import {
   transformShapes,
-  patternImport
+  transformShape,
+  polishVertices,
+  scaleImportedVertices
 } from './computer'
+import { getShape } from '../shapes/selectors'
 
 const getApp = state => state.app
 const getShapes = state => state.shapes
 const getTransforms = state => state.transforms
+const getCurrentShape = state => state.shapes.byId[state.shapes.currentId]
+const getTransform = state => state.transforms.byId[state.shapes.currentId]
 const getImporter = state => state.importer
-const getExporter = state => state.exporter
 const getMachine = state => state.machine
 
+// by returning null for shapes which can change size, this selector will ensure
+// transformed vertices are not redrawn when machine settings change
+const getTransformMachine = state => getTransform(state).canChangeSize ? null : state.machine
+
+// requires only shape, and in the case of erasers, machine state
+export const getShapedVertices = createSelector(
+  [
+      getCurrentShape,
+      getTransformMachine
+  ],
+  (shape, machine) => {
+    const state = {
+      shape: shape,
+      machine: machine
+    }
+    const metashape = getShape(shape)
+    return metashape.getVertices(state)
```

Comment:
This seems like the place to store the cache for the metashape base shapes. Storing a list of state.shapes to caches of metashape.getVertices(state), and then recalling them if they are in the cache.

I'm just not sure how to cache them in redux. Do we make another selector for the cache? Is that counter productive?

Reply:
This is related to #139